appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/util/ResourceValidator.java

```diff
@@ -0,0 +1,681 @@
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+
+package com.sun.enterprise.deployment.util;
+
+import com.sun.enterprise.config.serverbeans.Cluster;
+import com.sun.enterprise.config.serverbeans.Domain;
+import com.sun.enterprise.config.serverbeans.Server;
+import com.sun.enterprise.deployment.*;
+import org.glassfish.api.admin.ServerEnvironment;
+import org.glassfish.api.deployment.DeployCommandParameters;
+import org.glassfish.api.deployment.DeploymentContext;
+import org.glassfish.api.event.EventListener;
+import org.glassfish.api.event.Events;
+import org.glassfish.deployment.common.DeploymentException;
+import org.glassfish.deployment.common.JavaEEResourceType;
+import org.glassfish.internal.deployment.Deployment;
+import org.glassfish.logging.annotation.LogMessageInfo;
+import org.jvnet.hk2.annotations.Service;
+import org.glassfish.resourcebase.resources.api.ResourceConstants;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import java.net.MalformedURLException;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Created by Krishna Deepak on 6/9/17.
+ */
+@Service
+public class ResourceValidator implements EventListener, ResourceValidatorVisitor {
+
+    public static final Logger deplLogger = com.sun.enterprise.deployment.util.DOLUtils.deplLogger;
+
+    @LogMessageInfo(
+            message = "JNDI lookup failed for the resource with jndi name: {0}",
+            level = "SEVERE",
+            cause = "JNDI lookup for the specified resource failed.",
+            action = "Configure the required resources before deploying the application.",
+            comment = "For the method validateJNDIRefs of com.sun.enterprise.deployment.util.ResourceValidator."
+    )
+    private static final String RESOURCE_REF_JNDI_LOOKUP_FAILED = "AS-DEPLOYMENT-00026";
+
+    private String target;
+
+    private DeploymentContext dc;
+
+    private Application application;
+
+    @Inject
+    private Events events;
+
+    @Inject
+    private Domain domain;
+
+    public void postConstruct() {
+        events.register(this);
+    }
+
+    @Override
+    public void event(Event event) {
+        if (event.is(Deployment.AFTER_APPLICATION_CLASSLOADER_CREATION)) {
+            dc = (DeploymentContext) event.hook();
+            application = dc.getModuleMetaData(Application.class);
+            DeployCommandParameters commandParams = dc.getCommandParameters(DeployCommandParameters.class);
+            target = commandParams.target;
+            if (application == null || Boolean.getBoolean("deployment.switchoff.resource.validation"))
```

Comment:
Is this a temporary system property till the feature implementation/testing is complete?